### ISSUE TITLE
Error when loading very wide files

### ISSUE THREAD

**Issue by PRijnbeek** [OHDSI/WhiteRabbit#185]
We are working on databases (registries) than have 12000+ columns and they report that White Rabbit is not able to handle those.

"Exceeded maximum number parameters" error message.

They experimented with this that it seemed to work till 5000 columns.

This is of course an extreme number of columns but good to know why this happens and if there is a maximum we should make this clear in the documentation.

**Comment by PRijnbeek** [comment]
Update: they tried again on the latest version and the problem seems to be solved. 

Good to double check there is indeed no limit.

**Comment by PRijnbeek** [comment]
However, the report was empty... so not closing this yet

**Comment by MaximMoinat** [comment]
Thanks for reporting, was not aware of this limit. My guess it is a limit of an underlying data structure used. Could you share the full stacktrace of the error (an Error.txt should be created in the working folder)?

**Comment by MaximMoinat** [comment]
Also, even if the scan report succeeds, how would one work with this in Rabbit in a Hat?

Recently, @marcel1334 and I were actually discussing this. One conclusion we reached was that in these cases the source data might have to be transformed first into a more 'OMOP friendly' format. In this case that would be 'melting' the table from this wide to a long format. This would also greatly simplify the ETL implementation.

**Comment by PRijnbeek** [comment]
Yes I fully agree with that, we are handling this now outside RaH already, however white rabbit should not crash. In this case we use Usagi to map a column to a concept_id using the overview table of White Rabbit :)

I do not have the message myself, it was reported to me.

**Comment by MaximMoinat** [comment]
Btw: Excel is [limited to 16,384 columns](https://support.office.com/en-us/article/excel-specifications-and-limits-1672b34d-7043-467e-8e27-269d656771c3). In the scan report Table sheet (with value counts) we need one for the values, one for the frequencies. So the upper limit is a little over 8,000 columns per table. And apparently some Java code breaks before we reach that.

Will include this in a next update of the documentation. If you could get more information on the error, that would be welcome.
